dotnet-authserver/src/TeacherIdentity.AuthServer/Journeys/CheckJourneyTypeAttribute.cs

```diff
@@ -0,0 +1,35 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+
+namespace TeacherIdentity.AuthServer.Journeys;
+
+[AttributeUsage(AttributeTargets.Class, AllowMultiple = false)]
+public class CheckJourneyTypeAttribute : Attribute, IResourceFilter, IOrderedFilter
```

Comment:
This has to be an `IResourceFilter` rather than an `IPageFilter` so that it runs before the `PageModel` is constructed. If that weren't the case, `PageModel`s that have a specific type of `SignInJourney` in their constructor would blow up (due to the mis-matching journey type).